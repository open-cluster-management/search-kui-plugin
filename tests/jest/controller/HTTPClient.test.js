/*******************************************************************************
* Licensed Materials - Property of IBM
* (c) Copyright IBM Corporation 2019. All Rights Reserved.
*
* Note to U.S. Government Users Restricted Rights:
* Use, duplication or disclosure restricted by GSA ADP Schedule
* Contract with IBM Corp.
*******************************************************************************/
'use strict'
const queries = require('../../../dist/src-web/definitions/search-queries')
const HTTPClient = require('../../../dist/src-web/controller/HTTPClient')

const svc = 'svc'
const search = 'search'

describe('HTTP Client - POST', () => {
  const spy = jest.spyOn(HTTPClient, 'default')
  const _ = HTTPClient.default('post', search, queries.SAVED_SEARCH_QUERY).then(data => Promise.resolve(data))

  it(`return a axios request object with urlType - [${search}] - POST`, () => {
    expect(spy).toBeCalled()
    expect(HTTPClient.default).toMatchSnapshot()
  })
})


describe('HTTP Client - GET', () => {
  const spy = jest.spyOn(HTTPClient, 'default')
  const _ = HTTPClient.default('get', svc, undefined).then(data => Promise.resolve(data))

  it(`return a axios request object with urlType - [${svc}] - GET`, () => {
    expect(spy).toBeCalled()
    expect(HTTPClient.default).toMatchSnapshot()
  })
})
