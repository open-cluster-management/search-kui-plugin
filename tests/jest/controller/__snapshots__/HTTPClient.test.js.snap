// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTTP Client - GET return a axios request object with urlType - [svc] - GET 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "post",
      "search",
      Object {
        "operationName": "savedSearches",
        "query": "query savedSearches {
  items: savedSearches {
    id
    name
    description
    searchText
    __typename
  }
}
",
        "variables": Object {},
      },
    ],
    Array [
      "get",
      "svc",
      undefined,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`HTTP Client - POST return a axios request object with urlType - [search] - POST 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "post",
      "search",
      Object {
        "operationName": "savedSearches",
        "query": "query savedSearches {
  items: savedSearches {
    id
    name
    description
    searchText
    __typename
  }
}
",
        "variables": Object {},
      },
    ],
    Array [
      "get",
      "svc",
      undefined,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
